modules/deploy/main.tf

```
@@ -106,7 +109,7 @@ resource "local_file" "deploy_script" {
 }
 
 resource "null_resource" "deploy" {
-  count = var.create && var.create_deployment ? 1 : 0
+  count = var.run_deployment ? 1 : 0
```

```suggestion
  count = var.create && var.create_deployment && var.run_deployment ? 1 : 0
```

`run_deployment = true` should be used in addition to the rest.